src/Gui/Tree.cpp

```diff
@@ -3105,10 +3103,16 @@ bool DocumentItem::createNewItem(const Gui::ViewProviderDocumentObject& obj,
             pdata = std::make_shared<DocumentObjectData>(
                     this, const_cast<ViewProviderDocumentObject*>(&obj));
             auto &entry = getTree()->ObjectTable[obj.getObject()];
-            if(entry.size())
-                pdata->updateChildren(*entry.begin());
-            else
+            if(entry.size()) {

```

Comment:
I think this would read better as
```
            if(entry.size() > 0) {
```
or
```
            if(!entry.empty()) {
```
or flip the order of the conditions and use
```
            if(entry.empty()) {
```